src/main/utils/http.ts

```
@@ -0,0 +1,70 @@
+import { createWriteStream } from "original-fs";
+import { Readable } from "stream";
+import { finished } from "stream/promises";
+
+type Url = string | URL;
+
+export async function checkedFetch(url: Url, options?: RequestInit) {
+    try {
+        var res = await fetch(url, options);
+    } catch (err) {
+        if (err instanceof Error && err.cause) {
+            err = err.cause;
```

Biome (1.9.4)

[error] Reassigning a catch parameter is confusing.

The catch parameter is declared here:

Use a local variable instead.

(lint/suspicious/noCatchAssign)